### ISSUE TITLE
Convert backticks

### ISSUE THREAD

**Issue by IanVS** [sindresorhus/to-single-quotes#8]
With the arrival of ES2015, it would be useful if this library could also convert backticks to single quotes.

**Comment by IanVS** [comment]
I have a PR nearly ready for this, just need a few more tests.

Can you explain why there are still double quotes in this test:

```js
t.is(fn(JSON.stringify({a: '<a href="addr">'})), '{\'a\':\'<a href="addr">\'}');
```

My implementation is failing because it's converting them to single quotes.

**Comment by sindresorhus** [comment]
Why do you need this? What for? I don't see why anyone would want to switch all backticks to single-quotes. They have different behavior and you can't just convert them naively.

**Comment by IanVS** [comment]
The ESLint `"quotes"` rule currently allows backticks in all situations even when `"single"` or `"double"` is specified.  We are [changing](https://github.com/eslint/eslint/issues/3090) this behavior to only allow backticks when they could not be directly replaced with the specified quote type (you also suggested this in https://github.com/eslint/eslint/issues/4260#issuecomment-151204221).  The quotes rule is autofixable, so after we decide that backticks should be replaced with single or double quotes, we need a way to convert them.  Currently, we use this module to do the conversions, so I was hoping to also use it for replacing backticks.

**Comment by sindresorhus** [comment]
Ok, yeah, valid use-case, but I think it should be opt-in through an option here. `templateStrings: true` or something.

**Comment by sindresorhus** [comment]
> Can you explain why there are still double quotes in this test:

I don't remember exactly, but I would guess it's because we only convert top-level quotes.
It was introduced in https://github.com/sindresorhus/to-single-quotes/pull/2 by @shahata.

**Comment by IanVS** [comment]
> I think it should be opt-in through an option here.

:+1: Agreed, good idea.  I made a PR, but my RegExp skills are pretty weak so maybe there's a better way to accomplish this.  I can keep hacking on it if you'd like, and add an option. Of if you have a better way than I've proposed so far, please feel free to jump in.  :)

**Comment by shahata** [comment]
Regarding this test:
```js
t.is(fn(JSON.stringify({a: '<a href="addr">'})), '{\'a\':\'<a href="addr">\'}');
```
My understanding is that the rule is that you want to convert double quotes **enclosing** all javascript strings to single quotes, but you should **never** touch characters inside the string... I mean
```js
_.equals(eval(str), eval(toSingleQuotes(str)))
```
should always be true.

**Comment by IanVS** [comment]
@shahata, that's possible, but probably only applies when there are double quotes inside single quotes.  Otherwise, what would you do for the following string?

```js
var someText = "some test with nested 'single' quotes";
```

But yes, it makes sense to me that if the string is already wrapped in single quotes, to leave the whole thing alone.  Is that the intent, @sindresorhus?

**Comment by shahata** [comment]
Right, I meant never in the context of a JSON input, not string input.

**Comment by sindresorhus** [comment]
> But yes, it makes sense to me that if the string is already wrapped in single quotes, to leave the whole thing alone. Is that the intent,

Yes, it should only convert top-level quotes, and then escape as needed.

**Comment by sindresorhus** [comment]
This should be a lot simpler now that we're actually parsing the string: https://github.com/sindresorhus/to-single-quotes/commit/7f203d0f6a25ab5e74911732b77de02f8dc7eae2

**Comment by sindresorhus** [comment]
I dug into it and the template conversion path keeps growing into a partial JS parser with a lot of edge cases and regression risk. I want to keep this package small and predictable, so I am going to decline this feature.
